src/_guides/openapi/advanced-ref-usage.md

```diff
@@ -0,0 +1,335 @@
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Tic Tac Toe
+  description: |
+    This API allows writing down marks on a Tic Tac Toe board
+    and requesting the state of the board or of individual squares.
+  version: 1.0.0
+tags:
+  - name: Gameplay
+paths:
+  /board:
+    $ref: paths/board.yaml
+  /board/{row}/{column}:
+    $ref: paths/board_{row}_{column}.yaml
+```
+
+The `paths/board.yaml` looks like this:
+
+```yaml
+get:
+  summary: Get the whole board
+  description: Retrieves the current state of the board and the winner.
+  tags:
+    - Gameplay
+  operationId: get-board
+  responses:
+    '200':
+      description: OK
+      content:
+        application/json:
+          schema:
+            $ref: ../components/schemas/status.yaml
+```
+
+Finally, `components/schemas/status.yaml` looks like this:
+
+```yaml
+type: object
+properties:
+  winner:
+    $ref: ./winner.yaml
+  board:
+    $ref: ./board.yaml
+```
+
+All the filepaths are relative to their current location, and can traverse up and down the filesystem, with the standard `../` to go up a directory.
+
+The chance of getting a Git conflict when two different developers add two different paths or expanding properties in a schema is now a fair bit smaller. If a conflict does occur, the diff will be a lot less confusing to work out.
+
+One downside of splitting up components into different documents like this is that it becomes harder to follow API changes, either directly or by looking at files in GitHub. Changing a schema in one document can effect how multiple different endpoints work, and that can caused a bit of confusion. API change management tools like [Bump.sh](https://bump.sh/api-change-management) or [Optic](https://www.useoptic.com/docs/diff-openapi) can help by spotting breaking changes and reporting them on PRs, so that you can easily see problems that could otherwise slip through.
+
+## AsyncAPI Reusable Components
+
+AsyncAPI is thankfully the same when it comes to `$ref` and `components`, so if your event-driven API is struggling as much as your HTTP API then it's time to split things up.
+
+```yaml
+  v0/rust/servers/{server_id}/players/{steam_id}/events/banned:
+    description: Channel for notifying a server banned a player
+    parameters:
+      server_id:
+        "$ref": "./components/parameters.json#/server_id"
+      steam_id:
+        "$ref": "./components/parameters.json#/steam_id"
+    subscribe:
+      operationId: ServerPlayerBanned
+      message:
+        "$ref": "./components/messages/ServerPlayerBanned.json"
+```
+
+This example is taken from the [Gaming API](https://github.com/GamingAPI/definitions/) example projects, and highlights a slightly different approach of using a single `parameters.json` document and referencing a parameter within that file, instead of using a `parameters/` subdirectory with a file for each parameter. You could do either with either OpenAPI or AsyncAPI, it's a matter of personal preference. 
```

Comment:
I feel we need to add an extra sentence to explain the syntax of the ref value mixing both an external ref + internal ref.  `[external_file_path][internal_document_json_path]` where in your example `[external_file_path]="./components/parameters.json"` and `[internal_document_json_path]="#/server_id"`

Maybe the sentence, only lacks a concrete example when mentioning “referencing a parameter with that file (e.g. `...#/server_id`)”

WDYT?